core/src/main/java/org/infinispan/configuration/cache/IndexingConfigurationBuilder.java

```diff
@@ -169,9 +189,6 @@ public void validate() {
             //TODO  [anistor] This does not take into account eventual programmatically defined entity mappings
             CONFIG.noIndexableClassesDefined();
          }
-         if (attributes.attribute(INDEX).get() == Index.ALL && !clustering().cacheMode().isReplicated()) {
-            CONFIG.allIndexingInNonReplicatedCache();
```

Comment:
Why is this warning no longer relevant ?

Reply:
Since https://issues.redhat.com/browse/ISPN-11129, DIST caches should use ALL so that indexing can have redundancy. With this PR, Index.ALL is also deprecated, so this warning is twice outdated

Reply:
Then let's just deprecate the Index enum entirely and also not write the Index in the Searializer for now. The rest of the Serializer fixes can come in another PR. and I'll merge. Seems fine.

Reply:
done